### ISSUE TITLE
add "without category" in the "categorize selector" 

### ISSUE THREAD

**Issue by cloud2018** [nextcloud/cookbook#110]
To find recipes without category an additional item "without category" in the "categorze selector" dropdown would be helpful

In this way you could find recipes and add missing categories

**Comment by mrzapp** [comment]
@Teifun2 I came across this while testing:

```
An exception occurred while executing 'SELECT "r".* FROM "oc_cookbook_keywords" "k" INNER JOIN "oc_cookbook_names" "r" ON k.recipe_id = r.recipe_id WHERE (k.user_id = ?) AND (k.name = "") ORDER BY "k"."name" ASC' with params ["jeppe"]: SQLSTATE[42601]: Syntax error: 7 ERROR: zero-length delimited identifier at or near """" LINE 1: ... r.recipe_id WHERE (k.user_id = $1) AND (k.name = "") ORDER ... 
```

**Comment by Teifun2** [comment]
Oh, That's not good. I will look into that!

**Comment by mrzapp** [comment]
@Teifun2 I just tested against a random library of categorised and uncategorised recipes. I'll dig a little deeper, maybe I can uncover this.

**Comment by mrzapp** [comment]
@Teifun2 It seems to be a PostgreSQL specific issue, but still not sure how to work around it

**Comment by mrzapp** [comment]
@Teifun2 It's this line: 

```php
->andWhere('k.name = ""')
```
I replaced it with a generated expression like this:

```php
->andWhere($qb->expr()->isNull('k.name'))
```

...but the logic doesn't return any results, no matter the data set.

You might want to approach this differently, perhaps like this (pseudo query, probably not correct):

`SELECT recipe_id,name FROM recipe_names WHERE recipe_id NOT IN(SELECT recipe_id FROM recipe_keywords)`

**Comment by Teifun2** [comment]
@mrzapp I found out that Postgres seems to not like double quotes, so i swapped them around and it seemed to work!

`->andWhere('k.name = ""')` 

I will leave the Issue open until you can confirm that you also don't have any problems.

**Comment by mrzapp** [comment]
Yep, works on Postgres now :+1:

**Comment by cloud2018** [comment]
In Cookbook 0.7.5  it's not possible to find recipes without category, because "without category" is not displayed as folder:

In total there are 3 recipes, 2 with a category one recipe has no category

![cb1](https://user-images.githubusercontent.com/34692335/85797482-018a8f00-b73c-11ea-861c-63c42aa5536e.png)


**Versions**
Cookbook 0.7.5
Nextcloud 18.0.6

**Comment by seyfeb** [comment]
this was solved by #362
